lib/HtmlPhpExcel/HtmlPhpExcel.php

```diff
@@ -184,8 +184,12 @@ private function createExcel(): void
             // Loop over all rows
             $rowNumber = $this->getHighestRow($excelWorksheet);
             foreach($table->getRows() as $row){
+                $rowStylesArray = $this->getRowStylesArray($row);
+                if (!empty($rowStylesArray)) {
+                    $excelWorksheet->getStyle($rowNumber.':'.$rowNumber)->applyFromArray($rowStylesArray);
+                }
 
-                $excelWorksheet->getStyle($rowNumber.':'.$rowNumber)->applyFromArray($this->getRowStylesArray($row));
+                $excelWorksheet->getCell('A'.$rowNumber); // Fixes RowIterator exception
```

Comment:
What do you mean by `Fixes RowIterator exception`? When and why would that exception be thrown?

Reply:
It is (was) `PhpSpreadsheetException` thrown in `\PhpOffice\PhpSpreadsheet\Worksheet\RowIterator::resetStart()`.

`$startRow` was `2` and `$this->subject->getHighestRow()` returned `1`

I was surprised with that, but in `applyFromArray()` there are some `getCell()` calls that lead to cell/row creation.

Now I wonder if we can move `$this->setDimensionsForRow()` after `foreach($row->getCells()` loop, but it may lead to overwriting cells dimensions if some are set for a cell and for whole row :thinking: